src/util/bip32.rs

```diff
@@ -1080,5 +1111,50 @@ mod tests {
         assert_eq!("42", &format!("{}", ChildNumber::from_normal_idx(42).unwrap()));
         assert_eq!("000042", &format!("{:06}", ChildNumber::from_normal_idx(42).unwrap()));
     }
+
+    #[test]
+    fn schnorr_keypair() {
+        let secp = Secp256k1::new();
+        let xpriv = ExtendedPrivKey::from_str("xprv9s21ZrQH143K3QTDL4LXw2F7HEK3wJUD2nW2nRk4stbPy6cq3jPPqjiChkVvvNKmPGJxWUtg6LnF5kejMRNNU3TGtRBeJgk33yuGBxrMPHi").unwrap();
+        let pair = KeyPair::from_seckey_slice(&secp, &xpriv.private_key[..]).unwrap();
+        assert_eq!(xpriv.to_schnorr_keypair(&secp), pair);
+    }
+
+    #[test]
+    #[should_panic(expected = "Secp256k1(InvalidSecretKey)")]
```

Comment:
In ca4e90cd3eae74ad6f7a93af22aacf3eec53e73d:

I have a mild preference for using `unwrap_err` (and then maybe `to_string`) to compare errors inside the test, rather than using `should_panic`.